e2e-tests/portal/cypress/e2e/journeys/payment-report-officer/record-corrections/feature-flag-enabled/pending-corrections.spec.js

```diff
@@ -0,0 +1,196 @@
+        const nextDueReportPeriod = { start: { month: 1, year: 2022 }, end: { month: 2, year: 2022 } };
+        const formattedNextDueReportPeriod = getFormattedReportPeriodWithLongMonth(nextDueReportPeriod);
+
+        beforeEach(() => {
+          cy.task('getUserFromDbByEmail', BANK1_PAYMENT_REPORT_OFFICER1.email).then((user) => {
+            const { bank } = user;
+            const bankId = bank.id;
+
+            const report = UtilisationReportEntityMockBuilder.forStatus(REPORT_NOT_RECEIVED)
+              .withId(2)
+              .withReportPeriod(nextDueReportPeriod)
+              .withBankId(bankId)
+              .build();
+
+            cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+          });
+
+          cy.login(BANK1_PAYMENT_REPORT_OFFICER1);
+          cy.visit(relativeURL('/utilisation-report-upload'));
+        });
+
+        it('should display pending corrections', () => {
+          cy.assertText(mainHeading(), 'Report GEF utilisation and fees');
+
+          cy.assertText(pendingCorrections.correctionsHeading(), 'Record correction');
+          pendingCorrections.table().should('exist');
+
+          // Number of rows is number of corrections + 1 for the header
+          pendingCorrections.rows().should('have.length', 2);
+
+          cy.assertText(pendingCorrections.row(1).facilityId(), pendingCorrectionDetails.facilityId);
+          cy.assertText(pendingCorrections.row(1).exporter(), pendingCorrectionDetails.exporter);
+          cy.assertText(pendingCorrections.row(1).errorSummary(), pendingCorrectionDetails.additionalInfo);
+        });
+
+        it('should not display utilisation report file upload', () => {
+          utilisationReportUpload.utilisationReportFileInput().should('not.exist');
+        });
+
+        it('should display a message explaining when the next report can be uploaded to the user', () => {
+          pendingCorrections.nextReportDueHeading().should('exist');
+          cy.assertText(pendingCorrections.nextReportDueHeading(), `${formattedNextDueReportPeriod} report`);
+
+          pendingCorrections.nextReportDueText().should('exist');
+          cy.assertText(
+            pendingCorrections.nextReportDueText(),
+            `The ${formattedNextDueReportPeriod} report is due, but cannot be uploaded until the record corrections for the ${formattedPendingCorrectionReportPeriod} have been completed.`,
+          );
+
+          pendingCorrections.noReportDueHeading().should('not.exist');
+          pendingCorrections.noReportDueText().should('not.exist');
+        });
+      });
+
+      context('and when there are no reports due for upload', () => {
+        const formattedNextDueReportPeriodAlias = 'formattedNextDueReportPeriod';
+        const formattedUploadFromDateAlias = 'formattedUploadFromDate';
+
+        beforeEach(() => {
+          cy.task('getUserFromDbByEmail', BANK1_PAYMENT_REPORT_OFFICER1.email).then((user) => {
```

Comment:
could this be in a before and the login in a beforeEach so we are not doing this call for every it statement?

Reply:
I don't think so because I can't see a way to clean up that doesn't use an afterEach deleting everything without deleting specific entities which is not something we should do because that is harder to maintain and less reliable